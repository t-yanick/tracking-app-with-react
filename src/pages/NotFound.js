import React from 'react';
import { Link } from 'react-router-dom';
import notFoundImg from '../assets/images/notfound.png';

const NotFound = () => (
  <div className="notfound">
    <h1 className="heading">Not Found ... Sorry</h1>
    <div className="content">
      <img src={notFoundImg} alt="Not Found ... Sorry" className="image mb2" />
      <Link to="/" className="btn">Go back to Home</Link>
    </div>
  </div>
);

export default NotFound;
