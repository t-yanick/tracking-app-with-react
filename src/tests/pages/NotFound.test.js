import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NotFound from '../../pages/AddTrack';
import createTestStore from '../fixtures/createTestStore';

describe('NotFound', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <NotFound />
          </MemoryRouter>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
