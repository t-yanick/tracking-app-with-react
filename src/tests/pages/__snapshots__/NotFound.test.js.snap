// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should match with snapshot 1`] = `
<div
  className="add-track"
>
  <h1
    className="heading"
  >
    Add Track
  </h1>
  <div
    className="content"
  >
    <div
      className="content__msg"
    >
      Welcome back. Let's add your track for today!
    </div>
    <div
      className="track-form mb3"
    >
      
      <form
        className="track-form__form mb3"
        onSubmit={[Function]}
      >
        <div
          className="track-form__date"
        >
          <input
            onChange={[Function]}
            type="date"
            value="2021-10-04"
          />
        </div>
        <div
          className="track-form__group mb3"
        />
        <button
          className="btn dark"
          type="submit"
        >
          Save
        </button>
      </form>
    </div>
    
    
    <a
      className="btn"
      href="/tracks"
      onClick={[Function]}
    >
      Cancel & Back to Track List
    </a>
  </div>
</div>
`;
